### PR TITLE
Flood model improvement

### PR DESCRIPTION
**Pull request by SallyElHajjar**
I have added some improvements to the existing flood ML model.

### CODE REVIEW

File: usl_models/usl_models/flood_ml/model.py [hunk 1]
```diff
@@ -251,13 +251,16 @@ def __init__(
                 # Remaining layers are TimeDistributed and are applied to each
                 # temporal slice
                 layers.TimeDistributed(layers.Conv2D(8, 5, **st_cnn_params)),
+                layers.TimeDistributed(layers.BatchNormalization()),  # Batch Norm
```
Comment (tatianawu) [inline usl_models/usl_models/flood_ml/model.py:254]: Like I mentioned before, we don't use BatchNorm because the model is currently constrained to a batch size of 4 during training. During inference, we're often just predicting on 1-2 examples (depending on hardware). This is probably just going to add noise unless you can increase the batch size.

Is this a viable experiment? Yes. Is it what I think should be the priority right now? No.

IMO fixing data sparsity should take precedence over any other model work. Otherwise the model is going to continue producing zeros.

Comment (SallyElHajjar) [reply usl_models/usl_models/flood_ml/model.py:254]: it is evident that we should do other enhancements over the data pipeline too, i am not saying that this is just the only solution that we are going to proceed with. Regarding the batch norm, i had a plan to test also LayerNorm or GroupNorm, i believe that it's possible with proper tuning, it could still contribute to the stability and generalization of the model even with small batch size.

Comment (tatianawu) [reply usl_models/usl_models/flood_ml/model.py:254]: You can and should test this locally before checking it in then. Remember that you can train with local changes, so there's no reason to check in all the changes you're making for preliminary experiments.

I don't necessarily agree that "it's evident" you need to make model "improvements" before you've fixed data sparsity, because there's no telling what other performance issues may be present. Yes, there's very likely going to be room for improvement, but I don't see why that translates into immediately playing around with BatchNorm instead of doing things like longer horizon forecasting.

Comment (SallyElHajjar) [reply usl_models/usl_models/flood_ml/model.py:254]: i am saying it is evident for the data pipeline which includes the data sparsity handling, not the model improvement.

Comment (Katsutoshii) [reply usl_models/usl_models/flood_ml/model.py:254]: +1 to Tatiana's points. In general, I think before checking in model changes we should have data suggesting that this improves model performance.
